File: app/Models/Livraison.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Livraison extends Model
{
    protected $primaryKey = 'numLivraison';
    protected $fillable = [
        'numCommande',
        'numLieu',
        'transporteur',
        'referenceColis',
        'fraisLivraison',
        'contactTransporteur',
        'dateExpedition',
        'dateLivraison',
        'statutLivraison'
    ];

    public function commande()
    {
        return $this->belongsTo(Commande::class, 'numCommande');
    }

    public function lieu()
    {
        return $this->belongsTo(LieuLivraison::class, 'numLieu');
    }
}

File: app-mobile/lib/models/livraison.dart
class Livraison {
  final int numLivraison;
  final int numCommande;
  final String statutLivraison;
  final String? lieuLivraison;
  final String? transporteur;
  final String? referenceColis;
  final String? dateExpedition;
  final String? dateLivraison;

  Livraison({
    required this.numLivraison,
    required this.numCommande,
    required this.statutLivraison,
    this.lieuLivraison,
    this.transporteur,
    this.referenceColis,
    this.dateExpedition,
    this.dateLivraison,
  });

  factory Livraison.fromJson(Map<String, dynamic> json) {
    return Livraison(
      numLivraison: json['numLivraison'],
      numCommande: json['numCommande'],
      statutLivraison: json['statutLivraison'],
      lieuLivraison: json['lieuLivraison'],
      transporteur: json['transporteur'],
      referenceColis: json['referenceColis'],
      dateExpedition: json['dateExpedition'],
      dateLivraison: json['dateLivraison'],
    );
  }
}